static/src/javascripts/projects/common/modules/navigation/newHeaderNavigation.js

```diff
@@ -0,0 +1,152 @@
+// @flow
+
+import qwery from 'qwery';
+import fastdom from 'fastdom';
+import ophan from 'ophan/ng';
+import userAccount from 'common/modules/navigation/user-account';
+
+const html = qwery('html')[0];
+const menuItems = qwery('.js-close-nav-list');
+const enhanced = {};
+
+const weShouldEnhance = (checkbox: HTMLInputElement): ?boolean =>
+    !enhanced[checkbox.id] && checkbox && !checkbox.checked;
+
+const closeAllOtherPrimaryLists = (targetItem?: HTMLElement): void => {
+    menuItems.forEach(item => {
+        if (item !== targetItem) {
+            item.removeAttribute('open');
+        }
+    });
+};
+
+const removeOrderingFromLists = (): void => {
+    const mainListItems = qwery('.js-navigation-item');
+
+    mainListItems.forEach(item => {
+        const listItem = item;
+
+        listItem.style.order = '';
+    });
+};
+
+const toggleSidebar = (event: Event): void => {
+    // #? hacky type cast
+    const button: HTMLElement = (event.target: any);
+    const mainMenu = document.getElementById('main-menu');
+    const veggieBurgerLink = qwery('.js-change-link')[0];
+
+    const menuIsOpen = (): boolean =>
+        button.getAttribute('aria-expanded') === 'true';
+
+    if (!mainMenu || !veggieBurgerLink) {
+        return;
+    }
+    if (menuIsOpen()) {
+        fastdom.write(() => {
```

Comment:
How about moving these into function statements, so that they can be optimized? i.e. `fastdom.write(closeMenu)`